### PR TITLE
backported unirest removal

### PR DESCRIPTION
**Pull request by mehab**
### Description

Backported changes of https://github.com/DependencyTrack/dependency-track/issues/2350

### Checklist

- [x] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

File: src/main/java/org/dependencytrack/parser/vulndb/VulnDbParser.java [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+        }
+        if (result instanceof JSONObject) {
+            return this.parse((JSONObject) result, apiObject);
+        } else {
+            return this.parse((JSONArray) result, apiObject);
+        }
+    }
+
+    public <T> Results<T> parse(File file, Class<? extends ApiObject> apiObject) throws IOException {
+        String jsonData = Files.readString(Paths.get(file.toURI()), Charset.defaultCharset());
```
[the remark this answers is not in the document]
Comment (mehab) [reply src/main/java/org/dependencytrack/parser/vulndb/VulnDbParser.java:112]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/integrations/defectdojo/DefectDojoClient.java [hunk 2]
```diff
@@ -42,87 +50,97 @@ public class DefectDojoClient {
...
+        HttpEntity data = MultipartEntityBuilder.create().setMode(HttpMultipartMode.BROWSER_COMPATIBLE)
+                .addPart("file", inputStreamBody)
+                .addPart("engagement", new StringBody(engagementId, ContentType.MULTIPART_FORM_DATA))
+                .addPart("scan_type", new StringBody("Dependency Track Finding Packaging Format (FPF) Export", ContentType.MULTIPART_FORM_DATA))
+                .addPart("verified", new StringBody("true", ContentType.MULTIPART_FORM_DATA))
+                .addPart("active", new StringBody("true", ContentType.MULTIPART_FORM_DATA))
+                .addPart("minimum_severity", new StringBody("Info", ContentType.MULTIPART_FORM_DATA))
+                .addPart("close_old_findings", new StringBody("true", ContentType.MULTIPART_FORM_DATA))
+                .addPart("push_to_jira", new StringBody("push_to_jira", ContentType.MULTIPART_FORM_DATA))
+                .addPart("scan_date", new StringBody(DATE_FORMAT.format(new Date()), ContentType.MULTIPART_FORM_DATA))
```
[the remark this answers is not in the document]
Comment (mehab) [reply src/main/java/org/dependencytrack/integrations/defectdojo/DefectDojoClient.java:73]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/integrations/defectdojo/DefectDojoClient.java [hunk 3]
```diff
@@ -143,35 +161,38 @@ public ArrayList<String> jsonToList(final JSONArray jsonArray) {
...
+                .addPart("file", inputStreamBody)
+                .addPart("engagement", new StringBody(engagementId, ContentType.MULTIPART_FORM_DATA))
+                .addPart("scan_type", new StringBody("Dependency Track Finding Packaging Format (FPF) Export", ContentType.MULTIPART_FORM_DATA))
+                .addPart("verified", new StringBody("true", ContentType.MULTIPART_FORM_DATA))
+                .addPart("active", new StringBody("true", ContentType.MULTIPART_FORM_DATA))
+                .addPart("minimum_severity", new StringBody("Info", ContentType.MULTIPART_FORM_DATA))
+                .addPart("close_old_findings", new StringBody("true", ContentType.MULTIPART_FORM_DATA))
+                .addPart("push_to_jira", new StringBody("push_to_jira", ContentType.MULTIPART_FORM_DATA))
+                .addPart("test", new StringBody(testId, ContentType.MULTIPART_FORM_DATA))
+                .addPart("scan_date", new StringBody(DATE_FORMAT.format(new Date()), ContentType.MULTIPART_FORM_DATA))
```
[the remark this answers is not in the document]
Comment (mehab) [reply src/main/java/org/dependencytrack/integrations/defectdojo/DefectDojoClient.java:182]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/search/FuzzyVulnerableSoftwareSearchManager.java [hunk 4]
```diff
@@ -158,25 +168,25 @@ public SearchResult searchIndex(final String luceneQuery) {
...
+            NotificationUtil.dispatchExceptionNotifications(NotificationScope.SYSTEM, NotificationGroup.INDEXING_SERVICE, NotificationConstants.Title.CORE_INDEXING_SERVICES, content, NotificationLevel.ERROR);
         }
 
         indexManager.close();
         return searchResult;
     }
 
     private List<VulnerableSoftware> fuzzySearch(QueryManager qm, String luceneQuery) {
-        List<VulnerableSoftware>  fuzzyList = new LinkedList<>();
+        List<VulnerableSoftware> fuzzyList = new LinkedList<>();
```
[the remark this answers is not in the document]
Comment (mehab) [reply src/main/java/org/dependencytrack/search/FuzzyVulnerableSoftwareSearchManager.java]: @sonatype-lift ignoreall

File: src/main/java/org/dependencytrack/parser/vulndb/ModelConverter.java [hunk 5]
```diff
@@ -131,33 +135,33 @@ public static Vulnerability convert(final QueryManager qm, final us.springett.vu
         }
 
         CvssV2 cvssV2;
-        for (final CvssV2Metric metric : vulnDbVuln.getCvssV2Metrics()) {
-            cvssV2 = metric.toNormalizedMetric();
+        for (final org.dependencytrack.parser.vulndb.model.CvssV2Metric metric : vulnDbVuln.cvssV2Metrics()) {
```
[the remark this answers is not in the document]
Comment (mehab) [reply src/main/java/org/dependencytrack/parser/vulndb/ModelConverter.java]: @sonatype-lift ignoreall

**Review by sahibamittal — APPROVED** [review]
LGTM

**Pull request merged**